Scheduling/Infrastructure/EventStore/EsEventSerde.cs

```diff
@@ -48,7 +47,10 @@ public static T Deserialize<T>(this ResolvedEvent resolvedEvent)
     public static EventMetadata DeserializeMetadata(this ResolvedEvent resolvedEvent)
     {
         var jsonData = Encoding.UTF8.GetString(resolvedEvent.Event.Metadata.ToArray());
-        return JsonConvert.DeserializeObject<EventMetadata>(jsonData)!;
+        return JsonConvert.DeserializeObject<EventMetadata>(jsonData)! with
+        {
+            Position = resolvedEvent.Event.Position.CommitPosition
```

Comment:
**COMMENT:** We're deserialising first and then creating a new instance with a position from the event.